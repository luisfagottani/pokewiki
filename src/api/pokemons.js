import { fetchApi } from './api';

const Pokemons = {
  getAllPokemons: async () => {
    return fetchApi('pokemon', {
      method: 'GET',
    });
  },
};

export default Pokemons;
