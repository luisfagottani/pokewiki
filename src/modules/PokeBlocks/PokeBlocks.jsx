import PokeBlock from './components/PokeBlock';

import style from './PokeBlocks.module.scss';

const PokeBlocks = () => (
  <ul className={style.pokeblocks}>
    <PokeBlock />
    <PokeBlock />
    <PokeBlock />
    <PokeBlock />
    <PokeBlock />
  </ul>
);

export default PokeBlocks;
