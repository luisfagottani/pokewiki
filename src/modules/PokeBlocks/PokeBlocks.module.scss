.pokeblocks {
  // Box Model
  display: flex;
  flex-wrap: wrap;
  margin: 0px -15px;
}
