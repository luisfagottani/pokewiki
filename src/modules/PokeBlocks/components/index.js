import PokeBlock from './PokeBlock';

export default PokeBlock;
