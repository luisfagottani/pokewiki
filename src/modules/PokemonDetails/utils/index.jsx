const progressBarColor = (progress) => {
  if (progress < 60) {
    return 'blue';
  } else if (progress >= 60 && progress <= 70) {
    return 'yellow';
  } else {
    return 'red';
  }
};

export { progressBarColor };
