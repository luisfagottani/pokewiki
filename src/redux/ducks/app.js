import produce from 'immer';

export const REDUCER_NAME = 'app';

// TYPES
export const Types = {
  SET_GLOBAL_LOADING: `${REDUCER_NAME}/SET_GLOBAL_LOADING`,
};

// INITIAL STATE
const initialState = {
  isLoading: true,
  locale: 'pt-BR',
};

// REDUCER
export const reducer = produce((draft = initialState, action) => {
  switch (action.type) {
    case Types.SET_GLOBAL_LOADING:
      draft.isLoading = action.payload;
      return draft;

    case Types.SET_USER_PROFILE:
      draft.profile = action.profile;
      return draft;

    default:
      return draft;
  }
});

// ACTION FUNCTIONS

export function defineGlobalLoading(status) {
  return {
    type: Types.SET_GLOBAL_LOADING,
    payload: status,
  };
}

// SELECTOR FUNCTIONS
export function getGlobalLoading(state) {
  return state[REDUCER_NAME].isLoading;
}

export function getLocale(state) {
  return state[REDUCER_NAME].locale;
}
