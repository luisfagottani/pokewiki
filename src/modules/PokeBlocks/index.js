import PokeBlocks from './PokeBlocks';

export default PokeBlocks;
