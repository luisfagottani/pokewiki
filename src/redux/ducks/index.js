import { combineReducers } from 'redux';

import { reducer as AppReducer, REDUCER_NAME as APP_REDUCER_NAME } from './app';
import { reducer as PokemonsReducer, REDUCER_NAME as APP_POKEMONS_REDUCER } from './pokemons';
import { reducer as GenerationsReducer, REDUCER_NAME as APP_GENERATIONS_REDUCER } from './generations';

export default combineReducers({
  [APP_REDUCER_NAME]: AppReducer,
  [APP_POKEMONS_REDUCER]: PokemonsReducer,
  [APP_GENERATIONS_REDUCER]: GenerationsReducer,
});
