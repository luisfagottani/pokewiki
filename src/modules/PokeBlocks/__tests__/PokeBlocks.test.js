import React from 'react';
import { createMockStore, render } from 'utils/tests';
import { reducer as AppReducer, REDUCER_NAME as APP_REDUCER_NAME } from 'redux/ducks/app';
import { reducer as PokemonsReducer, REDUCER_NAME as POKEMON_REDUCER_NAME } from 'redux/ducks/pokemons';
import PokeBlocks from '..';

const mockStore = createMockStore({
  reducers: { [APP_REDUCER_NAME]: AppReducer, [POKEMON_REDUCER_NAME]: PokemonsReducer },
  initialState: {
    [APP_REDUCER_NAME]: {
      isLoading: true,
      locale: 'EN',
    },
    [POKEMON_REDUCER_NAME]: {
      species: { bulbasaur: { name: 'bulbasaur' }, ivysaur: { name: 'ivysaur' }, venusaur: { name: 'venusaur' } },
    },
  },
});

describe('<PokeBlocks />', () => {
  it('Should render without errors', () => {
    const { container } = render(<PokeBlocks />, mockStore);

    expect(container).toMatchSnapshot();
  });

  it('Should render with the corrects pokeblocks', () => {
    const { getByText } = render(<PokeBlocks />, mockStore);

    getByText('bulbasaur');
    getByText('ivysaur');
    getByText('venusaur');
  });
});
