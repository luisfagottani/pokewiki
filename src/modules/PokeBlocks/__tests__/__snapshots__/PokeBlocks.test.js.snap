// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokeBlocks /> Should render without errors 1`] = `
<div>
  <ul
    class="pokeblocks"
  >
    <li
      class="pokeblock"
    >
      <a
        class="pokeblock__content"
        href="/pokemon/bulbasaur"
      >
        <figure
          class="pokeblock__figure"
        >
          <img
            alt="bulbasaur"
            class="pokeblock__image"
            src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
          />
          <figcaption
            class="pokeblock__caption"
          >
            bulbasaur
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="pokeblock"
    >
      <a
        class="pokeblock__content"
        href="/pokemon/ivysaur"
      >
        <figure
          class="pokeblock__figure"
        >
          <img
            alt="ivysaur"
            class="pokeblock__image"
            src="https://img.pokemondb.net/artwork/ivysaur.jpg"
          />
          <figcaption
            class="pokeblock__caption"
          >
            ivysaur
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="pokeblock"
    >
      <a
        class="pokeblock__content"
        href="/pokemon/venusaur"
      >
        <figure
          class="pokeblock__figure"
        >
          <img
            alt="venusaur"
            class="pokeblock__image"
            src="https://img.pokemondb.net/artwork/venusaur.jpg"
          />
          <figcaption
            class="pokeblock__caption"
          >
            venusaur
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
